fluster/image/raw_image.py

```diff
@@ -0,0 +1,152 @@
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+# pylint: disable=too-many-instance-attributes, protected-access
+
+from pathlib import Path
+from typing import Dict, Optional, Tuple
+from fluster.image.raw_buffer import RawBuffer
+
+
+def get_full_meta_from_name(
+    filename: str, pixel_format: Optional[RawBuffer.Type] = None
+) -> Tuple[str, str, int, int, RawBuffer.Type, int]:
+    """Retrieves the metadata stored in the file name"""
+    filename_split = filename.split(".")
+    info = str(filename_split[0]).split("_")
+    video_name = info[0]
+    decode_type = info[1]
+    if pixel_format is None:
+        pixel_format = RawBuffer.Type(filename_split[1].lower())
+    width = int(info[2])
+    height = int(info[3])
+    num_frame = int(info[4])
+    return video_name, decode_type, width, height, pixel_format, num_frame
+
+
+def get_meta_from_name(
+    filename: str, pixel_format: Optional[RawBuffer.Type]
+) -> Tuple[str, int, int, RawBuffer.Type]:
+    """Retrieves part of the metadata stored in the file name"""
+    video_name, _, width, height, pixel_format, _ = get_full_meta_from_name(
+        filename, pixel_format
+    )
+    return video_name, width, height, pixel_format
+
+
+class RawImage:
+    """Class to handle the raw image"""
+
+    def __init__(
+        self,
+        buffer: bytes,
+        video_name: str,
+        width: int,
+        height: int,
+        pixel_format: RawBuffer.Type,
+        create_diff_buffer: bool = False,
+    ) -> None:
+
+        self._pixel_format: RawBuffer.Type = pixel_format
+        self._width: int = width
+        self._height: int = height
+        self._video_name: str = video_name
+        self._create_diff_buffer: bool = create_diff_buffer
+        self._data: Dict[RawBuffer.Type, RawBuffer] = {
+            pixel_format: RawBuffer(
+                buffer,
+                width,
+                height,
+                pixel_format,
+                create_diff_buffer=create_diff_buffer,
+            )
+        }
+
+    @staticmethod
+    def from_filename(
+        filename: str, pixel_format: Optional[RawBuffer.Type] = None
+    ) -> "RawImage":
+        """Create a RawImage from a file using the content and the metadata stored in the file name"""
+        file_path = Path(filename)
+        if not file_path.exists():
+            raise RuntimeError(f"The file {filename} does not exists")
+        buffer = bytes(file_path.read_bytes())
+        return RawImage(buffer, *get_meta_from_name(filename, pixel_format))
+
+    @staticmethod
+    def from_buffer(
+        filename: str,
+        buffer: bytes,
+        pixel_format: RawBuffer.Type,
+    ) -> "RawImage":
+        """Create a RawImage from a buffer using the content and the metadata stored in the file name"""
+        return RawImage(buffer, *get_meta_from_name(filename, pixel_format))
+
+    def __str__(self) -> str:
```

Comment:
Are you overloading the str() method?

Reply:
yes, but this only applies in order to be able to debug with better information when printing the object